var/spack/repos/builtin/packages/py-async-timeout/package.py

```diff
@@ -10,7 +10,13 @@ class PyAsyncTimeout(PythonPackage):
     homepage = "https://github.com/aio-libs/async-timeout"
     url      = "https://github.com/aio-libs/async-timeout/archive/v3.0.1.tar.gz"
 
+    version('4.0.2', sha256='2163e1640ddb52b7a8c80d0a67a08587e5d245cc9c553a74a847056bc2976b15')
+    version('4.0.1', sha256='b930cb161a39042f9222f6efb7301399c87eeab394727ec5437924a36d6eef51')
+    version('4.0.0', sha256='2116e8c7412929579e1d4e1b3c5fdfe3835c2002a0189451d183148239c05a17')
```

Comment:
It looks like you added the checksums for the PyPI tarball, not the GitHub tarball. This is breaking the package. Will submit a PR to fix this.

Reply:
I added what `spack checksum` gave me. If that's not correct, then the checksum function has an issue...

Reply:
If you run spack checksum again do you get these same checksums?

Reply:
With the current develop branch, it downloads the tarball from github and gives me checksums that are both different from yours and different from my original PR. I'm not sure what's happening. I suppose your PR fixes that by relying exclusively on PyPi rather than Github?

Reply:
I think what's happening is that some versions are downloading from:
https://github.com/aio-libs/async-timeout/releases/download/v4.0.2/async-timeout-4.0.2.tar.gz
while other versions are downloading from:
https://github.com/aio-libs/async-timeout/archive/refs/tags/v4.0.2.tar.gz

Currently, Spack checks the downloads page for new releases and assumes a unique URL for each version and archive (.tar.gz, .zip, etc.). For services like GitHub where there are multiple download options for the same version, Spack doesn't know what to do and randomly chooses one or the other.

So in this case, switching to PyPI will solve this issue for this package.